src/Controls/tests/DeviceTests/Elements/ContentView/ContentViewTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Handlers;
+using Microsoft.Maui.Hosting;
+using Microsoft.Maui.Platform;
+using Xunit;
+
+namespace Microsoft.Maui.DeviceTests
+{
+	[Category(TestCategory.ContentView)]
+	public partial class ContentViewTests : HandlerTestBase
+	{
+		void SetupBuilder()
+		{
+			EnsureHandlerCreated(builder =>
+			{
+				builder.ConfigureMauiHandlers(handlers =>
+				{
+					handlers.AddHandler<Label, LabelHandler>();
+					handlers.AddHandler<IContentView, ContentViewHandler>();
+					handlers.AddHandler<Grid, LayoutHandler>();
+				});
+			});
+		}
+
+		[Fact("ContentView updating it's ControlTemplate works")]
+		public async Task ControlTemplateUpdates()
+		{
+			SetupBuilder();
+			var child = new Label { Text = "Content 1" };
+			var contentView = new Microsoft.Maui.Controls.ContentView();
+			var header = new Label { Text = "Header" };
+			var footer = new Label { Text = "Footer" };
+			var presenter = new ContentPresenter();
+			var grid = new Grid();
+
+			var contentViewHandler = await CreateHandlerAsync<ContentViewHandler>(contentView);
+
+			await InvokeOnMainThreadAsync(() =>
+			{
+				contentView.Content = child;
+				Assert.True(GetChildCount(contentViewHandler) == 1);
```

Comment:
Does it makes sense to add a call to `GetContentChildCount` here? so that we're validating that the `GetContentChildCount` has indeed changed from here to the check later on?

Reply:
Good idea, fixed